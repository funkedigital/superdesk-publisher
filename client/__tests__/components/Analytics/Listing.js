import React from "react";
import Listing from "../../../components/Analytics/Listing";
import { render } from "@testing-library/react";

jest.mock("react-virtualized/styles.css", () => {
  return {};
});

const articles = {
  items: [],
  page: 0,
  totalPages: 1,
  loading: false,
  itemSize: 50
};

describe("Analytics/Listing", () => {
  it("renders correctly", async () => {
    const { container } = render(
      <Listing articles={articles}
        loading={false}
        queryArticles={jest.fn()} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
