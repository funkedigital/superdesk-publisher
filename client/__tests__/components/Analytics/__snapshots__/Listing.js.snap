// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics/Listing renders correctly 1`] = `
<div
  class="sd-column-box__main-column-inner sd-d-flex"
>
  <div
    class="sd-flex-table sd-table--shadowed sd-table--action-hover"
    style="flex-grow: 1; position: relative;"
  >
    <div
      class="sd-flex-table__row sd-flex-table--head"
    >
      <div
        class="sd-flex-table__cell sd-flex-grow"
      >
        <div>
          Title
        </div>
      </div>
      <div
        class="sd-flex-table__cell"
      >
        Â 
      </div>
      <div
        class="sd-flex-table__cell"
      >
        <div>
          Page Views
        </div>
      </div>
    </div>
    <p
      style="padding: 1em; text-align: center;"
    >
      No results found
    </p>
    <div
      style="overflow: visible; height: 0px; width: 0px;"
    >
      <div
        aria-label="grid"
        aria-readonly="true"
        class="ReactVirtualized__Grid ReactVirtualized__List"
        role="grid"
        style="box-sizing: border-box; direction: ltr; height: -46px; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
        tabindex="0"
      />
    </div>
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;
